### DISCUSSION TITLE
Status of Deprecated Methods/Properties

### DISCUSSION THREAD

**Discussion by GCuser99** [GCuser99/SeleniumVBA#86] [Announcements]
In the release notes of versions 4.3 and 4.4, I enumerated a number of methods/properties that now have improved alternatives. I'm posting this to give anyone who cares an opportunity to voice their opinion as to what to do with these deprecated methods/properties. We could just leave the overlapping functionality as it exists now, eliminate some or all of the deprecated ones, and/or convert at least several of these to short-cut the new methods (for example, convert WebDriver.SwitchToNewWindow to be equivalent to WebDriver.Windows.SwitchToNew).

I'll leave this issue open for a few weeks. Unless you are indifferent, please let me know what you think. **If I don't hear from anyone, my leaning is to just remove all of the deprecated methods**.

Here is the complete list below:

|Deprecated Method\Property|Class|Replacement Method\Property|Class|Example Usage|
| ---------------- | ------------- | ------------- |------------- |------------------------------- |
|ControlKey|WebKeyboard|CtrlKey|WebKeyboard|elem.SendKeys keys.CtrlKey & "a"|
|GetWindowHandles|WebDriver|Handles|WebWindows|Set handles = driver.Windows.Handles|
|SwitchToNewWindow|WebDriver|SwitchToNew|WebWindows|driver.Windows.SwitchToNew|
|SwitchToWindow|WebDriver|Item().Activate|WebWindows|windows(1).Activate|
|GetTitle|WebDriver|Title|WebWindow|Debug.Print driver.ActiveWindow.Title|
|GetCurrentWindowHandle|WebDriver|Handle|WebWindow|handle = driver.ActiveWindow.Handle|
|CloseWindow|WebDriver|CloseIt|WebWindow|window.CloseIt|
|GetWindowRect|WebDriver|Bounds|WebWindow|Set rect = driver.ActiveWindow.Bounds|
|SetWindowRect|WebDriver|Bounds|WebWindow|Set window.Bounds = rect|
|SetWindowSize|WebDriver|SetSize|WebWindow|driver.ActiveWindow.SetSize width:=500|
|SetWindowPosition|WebDriver|SetPosition|WebWindow|window.SetPosition x:=200, y:=200|
|MaximizeWindow|WebDriver|Maximize|WebWindow|driver.ActiveWindow.Maximize|
|FullScreenWindow|WebDriver|FullScreen|WebWindow|window.FullScreen|
|MinimizeWindow|WebDriver|Minimize|WebWindow|driver.Windows.SwitchToNext.Minimize|
|AlertAccept|WebDriver|Accept|WebAlert|driver.SwitchToAlert.Accept|
|AlertDismiss|WebDriver|Dismiss|WebAlert|driver.SwitchToAlert.Dismiss|
|GetAlertText|WebDriver|GetText|WebAlert|prompt = driver.SwitchToAlert.GetText|
|SetAlertText|WebDriver|SendKeys|WebAlert|driver.SwitchToAlert.SendKeys "my text"|
|SetFocusNoScroll|WebDriver|SetFocus|WebDriver|driver.SetFocus element:=elem, preventScroll:=True|
|SetFocusNoScroll|WebElement|SetFocus|WebElement|elem.SetFocus preventScroll:=True|

Notes:
- The Handles property of the WebWindows class returns a Collection, while the deprecated alternative returns a String Array
- The Item (default) method of the WebWindows class will take either an index or handle as input
- The SwitchToAlert method of the WebDriver class will wait up to a maximum amount of time (default 10 secs) for the Alert to show and then return a WebAlert object for further interaction
- The IsAlertPresent method of WebDriver class, not listed above, has no other functional equivalent so will keep as is

**Comment by 6DiegoDiego9** [comment]
I'm personally already OK with the removal of the old methods and properties. I don't have a big database to update.
Thanks for the very useful correspondence table!

**Reply by kwirth72** [reply]
Please update the following Wiki page to reflect the replacement of "ControlKey" by "CtrlKey" (and maybe the other replacements). Thank you!
https://github.com/GCuser99/SeleniumVBA/wiki

**Comment by GCuser99** [comment]
Done - thx!
